generator.yaml

```diff
@@ -621,6 +621,56 @@ resources:
         from:
           operation: DescribeFeatureGroup
           path: FeatureGroupStatus
+  NotebookInstance:
+    exceptions:
+      errors:
+          404:
+            code: ValidationException
+            message_prefix: RecordNotFound
+      terminal_codes:
+        - ResourceLimitExceeded
+        - ResourceNotFound
+        - ResourceInUse
+        - OptInRequired
+        - InvalidParameterCombination
+        - InvalidParameterValue
+        - MissingParameter
+        - MissingAction
+        - InvalidClientTokenId
+        - InvalidQueryParameter
+        - MalformedQueryString
+        - InvalidAction
+        - UnrecognizedClientException
+    hooks:
```

Comment:
Note to team (todo): we should ideally arrange these hooks in some logical order across the various resources.